infra/resources/_modules/functions_apps/locals.tf

```diff
@@ -0,0 +1,34 @@
+locals {
+
+  resource_group_name_common = "${var.project}-rg-common"
+  vnet_name_common           = "${var.project}-vnet-common"
+
+  resource_group_name_common_legacy = "${var.project_legacy}-rg-common"
+  vnet_name_common_legacy           = "${var.project_legacy}-vnet-common"
+
+  function_wallet = {
+
+
+    app_settings_common = {
+      FUNCTIONS_WORKER_RUNTIME       = "node"
+      WEBSITE_RUN_FROM_PACKAGE       = "1"
+      WEBSITE_DNS_SERVER             = "168.63.129.16"
+      FUNCTIONS_WORKER_PROCESS_COUNT = 4
+      NODE_ENV                       = "production"
+
+      // Keepalive fields are all optionals
+      FETCH_KEEPALIVE_ENABLED             = "true"
+      FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+      FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+      FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+      FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+      FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+      // CosmosDB connection
+      COSMOSDB_URI           = var.cosmos_db.endpoint
+      COSMOSDB_KEY           = var.cosmos_db.primary_key
```

Comment:
can we use managed identity to access database?

Reply:
here's an example https://github.com/pagopa/io-fims/blob/main/infra/resources/prod/_modules/web_apps/app_service_openid_provider.tf#L36

Reply:
correct we can define also custom roles for single container
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_sql_role_assignment

Also you need to change application code to use managed identity authentication (cc @lucacavallaro)

Reply:
@christian-calabrese recently set up AI Search to access Cosmos DB. The same config could be used here as well

Reply:
If one of the [built in cosmosdb roles](https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac#built-in-role-definitions) suits your case, you can also use one of them and reduce the scope to the database or collection level as done [here](https://github.com/pagopa/io-services-cms/blob/ebc186c93cf0966a2dc4f21ee138375dfb45f908/infra/_modules/ai_search/main.tf#L78C1-L84C2)

Reply:
Done in https://github.com/pagopa/io-wallet/pull/56/commits/a06686432f7f71d00738cc8e9e00e10a17b312d3

The `DocumentDB Account Contributor` should be right (see https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/databases#documentdb-account-contributor).

Reply:
yes `DocumentDB Account Contributor` is right.
You can use custom role to give more restrictive access (es. read-only a container)